aya/src/bpf.rs

```diff
@@ -1068,6 +1068,29 @@ impl Ebpf {
     pub fn programs_mut(&mut self) -> impl Iterator<Item = (&str, &mut Program)> {
         self.programs.iter_mut().map(|(s, p)| (s.as_str(), p))
     }
+
+    /// Enable global statistics tracking for all programs and returns a file descriptor handler.
+    ///
+    /// Statistics tracking will be disabled once the file descriptor is closed or released.
+    ///
+    /// Typical usage:
+    /// 1. Obtain fd from `enable_stats_fd` and bind it to a variable.
+    /// 2. Record the statistic of interest.
+    /// 3. Wait for a recorded period of time.
+    /// 4. Record the statistic of interest again, and calculate the difference.
+    /// 5. Close/release fd.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// # use aya::{Ebpf, EbpfError};
+    /// # use aya_obj::BpfStatsType;
+    /// let _fd = Ebpf::enable_stats_fd(BpfStatsType::RunTime)?;
+    /// # Ok::<(), EbpfError>(())
+    /// ```
+    pub fn enable_stats_fd(stats_type: BpfStatsType) -> Result<OwnedFd, SyscallError> {
```

Comment:
What if they add more tho? Presumably if they have an enum and not a boolean they plan to add more. Then we must break API. If we add a #[non_exhaustive] `Stats` then we don't have to break it.

Reply:
It's been this way for quite a long time though 🤔 I don't have strong opinions either way... just with my user hat on enable_stats(BpfStats::Runtime) feels weird when there is no other possible variant

Reply:
I agree it does feel a bit weird when the param will be the same every time, but I think I'm currently leaning towards explicitly requiring the `stats_type` param. For the reason that alessandrod mentioned, in case they do expand on `bpf_stats_type`.